rclpy/src/rclpy/node.hpp

```diff
@@ -17,157 +17,161 @@
- * Raises AttributeError if pyparameter_cls doesn't have a 'Type' attribute
- * Raises RuntimeError if assumptions about rcl structures are violated
- *
- * \param[in] pyparameter_cls The rclpy.parameter.Parameter class object.
- * \param[in] node_capsule Capsule pointing to the node handle
- * \return A dict mapping parameter names to rclpy.parameter.Parameter (may be empty).
- */
-py::dict
-get_node_parameters(py::object pyparameter_cls, py::capsule pynode);
-
-/// Create a node
-/**
- * Raises ValueError if the node name or namespace is invalid
- * Raises RCLError if the node could not be initialized for an unexpected reason
- * Raises RCLInvalidROSArgsError if the given CLI arguments could not be parsed
- * Raises UnknownROSArgsError if there are unknown CLI arguments given
- * Raises MemoryError if memory could not be allocated for the node
- *
- * \param[in] node_name name of the node to be created
- * \param[in] namespace namespace for the node
- * \param[in] pycontext Capsule for an rcl_context_t
- * \param[in] pycli_args a sequence of command line arguments for just this node, or None
- * \param[in] use_global_arguments if true then the node will also use cli arguments on pycontext
- * \param[in] enable rosout if true then enable rosout logging
- * \return Capsule of the pointer to the created rcl_node_t * structure
- */
-py::capsule
-create_node(
-  const char * node_name,
-  const char * namespace_,
-  py::capsule pycontext,
-  py::object pycli_args,
-  bool use_global_arguments,
-  bool enable_rosout);
-}  // namespace rclpy
+class Node : public Destroyable, public std::enable_shared_from_this<Node>
+{
+public:
+  /// Create a node
+  /**
+   * Raises ValueError if the node name or namespace is invalid
+   * Raises RCLError if the node could not be initialized for an unexpected reason
+   * Raises RCLInvalidROSArgsError if the given CLI arguments could not be parsed
+   * Raises UnknownROSArgsError if there are unknown CLI arguments given
+   * Raises MemoryError if memory could not be allocated for the node
+   *
+   * \param[in] node_name name of the node to be created
+   * \param[in] namespace namespace for the node
+   * \param[in] pycontext Capsule for an rcl_context_t
+   * \param[in] pycli_args a sequence of command line arguments for just this node, or None
+   * \param[in] use_global_arguments if true then the node will also use cli arguments on pycontext
+   * \param[in] enable rosout if true then enable rosout logging
+   * \return Capsule of the pointer to the created rcl_node_t * structure
+   */
+  Node(
+    const char * node_name,
+    const char * namespace_,
+    Context & context,
+    py::object pycli_args,
+    bool use_global_arguments,
+    bool enable_rosout);
+
+  /// Get the fully qualified name of the node.
+  /**
+   * Raises RCLError if name is not set
+   *
+   * \return String containing the fully qualified name of the node
+   */
+  const char *
+  get_node_fully_qualified_name();
```

Comment:
@ahcorde nit: do we need that `node_` token? Isn't it implied when doing `node.get_fully_qualified_name()`? Same for other APIs below.